state/parsers/dataTrieLeafParser.go

```diff
@@ -37,7 +37,7 @@ func (tlp *dataTrieLeafParser) ParseLeaf(trieKey []byte, trieVal []byte) (core.K
 	if tlp.enableEpochsHandler.IsAutoBalanceDataTriesEnabled() {
 		data := &dataTrieValue.TrieLeafData{}
 		err := tlp.marshaller.Unmarshal(data, trieVal)
-		if err == nil {
+		if err == nil && !isEmptyTrieData(data) {
```

Comment:
Please add a TODO to remove this after we have a proper fix on the marshaller.

Reply:
Done.